@lingosta/app/components/Translate/TranslateContainer.tsx

```diff
@@ -103,13 +106,19 @@ const TranslateContainer: React.FC = () => {
 
   useEffect(() => {
     userTranslationsList();
-    if (activeGroup) {
-      setisGroupModalOpen(false);
-    } else {
-      setisGroupModalOpen(true);
-    }
+    
+    const shouldShowGroupModal = !activeGroup;
+    setisGroupModalOpen(shouldShowGroupModal);
   }, [user, activeGroup]);
 
+  if(isLoading) {
+    return (
+      <div className='px-12 animate-pulse mx-auto items-center text-center py-8 font-bold text-xl'>
```

Comment:
Wrap this Loading div in its own component and use that one instead of redeclaring it multiple times.